### PR TITLE
Logcat API

### PR DESCRIPTION
**Pull request by bcallebaut**
This is the starting point for the abstraction of the logcat functionality.

## Implemented functionality
- Run the shell command : _exec logcat_ 
- Returns the stream of text from the logcat as a raw InputStream or as a Buffered Reader

## Not (yet) implemented features
- Filters
- Log records... or any other higher level API to access the Logs.
- Other functionalities provided by ddmlib

### CODE REVIEW

File: src/se/vidstige/jadb/Transport.java [hunk 1]
```diff
@@ -4,7 +4,7 @@
 import java.net.Socket;
 import java.nio.charset.Charset;
 
-class Transport {
+class Transport implements AutoCloseable{
```
Comment (vidstige) [inline src/se/vidstige/jadb/Transport.java]: 👍

Comment (vidstige) [reply src/se/vidstige/jadb/Transport.java]: But fix whitespace before brace (`{`)

**Comment by vidstige** [comment]
The codecov checks are not happy because you've added code without tests. Perhaps we can add some tests to the new code? Also please double check whitespace in all added/modified lines.

Let me know if you've need some help/ideas/whatever with getting the tests up. 

One idea here, is perhaps you want to make pr for just the refactoring of the public/package inconsistency? That's looks really good and ready to merge right away.

Again, big thanks for looking into this. I'm trying to be quick on reviewing to not have you waiting. Please feel free to mail me if I'm not reviewing in time, as I might miss a notification here and there! :)

**Comment by vidstige** [comment]
Closing this for now. Let me know if you need any pointers moving forward, or if you desire someone else to take over. :-)
